### PR TITLE
Add DBConnection.available_connection_options/0

### PR DESCRIPTION
**Pull request by whatyouhide**
Follow up to #290.

### CODE REVIEW

**Comment by josevalim** [comment]
`available_connection_options` perhaps?

**Comment by josevalim** [comment]
It seems most of our specs use `Keyword.t()`? Should we align that instead then? But if we are doing that, I would prefer to call it `t:DBConnection.connection_options` to avoid ambiguity?

**Comment by whatyouhide** [comment]
@josevalim yeah, the type is there but not really used. If you want, we can add `t:DBConnection.connection_option/0`, deprecate `t:DBConnection.option/0`, and change specs to use `DBConnection.connection_option/0`. Thoughts?

**Comment by josevalim** [comment]
Sure!

File: lib/db_connection.ex [hunk 1]
```diff
@@ -134,12 +134,30 @@ defmodule DBConnection do
...
+  > #### Deprecated {: .warning}
+  >
+  > This option is deprecated since v2.6.0. Use `t:connection_option/0` instead.
+
+  """
   @type option ::
           {:log, (DBConnection.LogEntry.t() -> any) | {module, atom, [any]} | nil}
           | {:queue, boolean}
           | {:timeout, timeout}
           | {:deadline, integer | nil}
```
Comment (josevalim) [inline lib/db_connection.ex:149]:
```suggestion
  @type option :: connection_option
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**